manifests/init.pp

```diff
@@ -494,12 +528,13 @@
       enable     => $trap_service_enable_real,
       hasstatus  => $trap_service_hasstatus,
       hasrestart => $trap_service_hasrestart,
-      require    => [
-        File['var-net-snmp'],
-        Package['snmptrapd'],
-      ],
+      require    => File['var-net-snmp'],
       subscribe  => File['snmptrapd.conf'],
     }
+
+    if $manage_packages {
+      Package['snmptrapd'] -> Service['snmptrapd']
```

Comment:
This would be easier to understand if the package just had a `before` attribute.

Reply:
While I agree, I was following the style of the existing code, rather than turning this PR into a gigantic refactor.